lib/build.ml

```diff
@@ -91,15 +96,22 @@ module Op = struct
       (Fmt.strf "@.\
                  To reproduce locally:@.@.\
                  %a@.\
+                 git fetch origin master@.\
+                 git merge %s@.\
                  cat > Dockerfile <<'END-OF-DOCKERFILE'@.\
                  \o033[34m%a\o033[0m@.\
                  END-OF-DOCKERFILE@.\
                  docker build .@.@."
          Current_git.Commit_id.pp_user_clone (Current_git.Commit.id commit)
+         master
          Dockerfile.pp (make_dockerfile ~for_user:true));
     let dockerfile = Dockerfile.string_of_t (make_dockerfile ~for_user:false) in
     Current.Job.start ~timeout:build_timeout ~pool job ~level:Current.Level.Average >>= fun () ->
-    Current_git.with_checkout ~enable_submodules:false ~job commit @@ fun dir ->
+    Current_git.with_checkout ~job commit @@ fun dir ->
+    (* Merge with master. The merge should work because we already tried this in the analysis step. *)
+    let cmd = "", [| "git"; "merge"; master |] in
```

Comment:
does this really work? I remember trying something like this but it doesn't work if the branch tested is older than the master branch as `Current_git.with_checkout` only fetches one branch at a time. Has this changed or was I missing something when I tried this back then?

Reply:
Oh, I would also recommend to use the `-q` argument to the git merge

Reply:
Good point. My test-case is indeed simply ahead of master. I guess a `git fetch origin master` first would fix that (pulling from the local clone).

Reply:
OK, I tested this and it does work as is. You have to use `git clone --single-branch` to disable that behaviour. The `master` hash comes from a separate fetch operation, so we know the repository we're cloning from already has it.

I've added `-q` as you suggested.